apps/ssr-tests-v9/src/utils/visitPage.test.ts

```diff
@@ -0,0 +1,32 @@
+import type { Page } from 'puppeteer';
+import { visitUrl } from './visitPage';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
+
+describe('visitUrl', () => {
+  it('calls .goto() 5 times before a failure', async () => {
+    expect.assertions(2);
+
+    jest.spyOn(console, 'warn').mockImplementation(noop);
+    jest.spyOn(console, 'error').mockImplementation(noop);
+
+    const pageMock: Partial<Page> = {
+      goto: jest.fn().mockImplementation(() => Promise.reject()),
+    };
+
+    await expect(visitUrl(pageMock as Page, 'https://localhost:8080')).rejects.toMatchInlineSnapshot(`undefined`);
```

Comment:
can we make it more explicit where that `undefined` comes from ?

```suggestion
    await expect(visitUrl(pageMock as Page, 'https://localhost:8080')).rejects.toMatchInlineSnapshot(`page wont open - mock`);
```